### ISSUE TITLE
About VQ/MLP/VIT training

### ISSUE THREAD

**Issue by waltstephen** [X-Omni-Team/X-Omni#14]
Hello, I have a question. If we take the vit vq mlp llm path, and train the vq mlp vit directly on the llava training mlp data while keeping the llava frozen, the trained llava seems to think the index is an invalid token and refuses to answer. However, Genji llava can achieve some visual capabilities if we only train the mlp with the llava frozen. So, is there something wrong with my training? [Sigh R]

In the vit->vq->mlp->llm path used in your paper, in the first stage of training, do you directly open the gradient of vit, vq, and mlp while keeping LLM frozen, allowing the downstream qwen2.5-1.5blllml to gain visual capabilities? Can we do vqa tasks without training the LLM part?

**Comment by zhangxiaosong18** [comment]
We didn't freeze the LLM, only training the LLM makes visual understanding possible

**Comment by waltstephen** [comment]
As I understand it, according to llava's native training, training only the MLP layer in the first stage yields some visual performance in frozen LLM. However, when I replicated your code and only trained the MLP, VQ, and VIT layers, I didn't see any improvement in the LLM visual performance. Do you mean that when training VQ, your VIT, VQ, MLP, and LLM layers all use gradients? Do they share a common learning rate? If you can answer I'd be very grateful.

**Comment by waltstephen** [comment]
Or did you use a similar training process as llava to train VQ+MLP first, then train VQ+MLP+LLM (qwen2.5-1.5B) for VQA? Or did you train all of them together?

**Comment by zhangxiaosong18** [comment]
We used a training method similar to llava, but added VQ

**Comment by waltstephen** [comment]
Thank you for your answer. I have another question. In the first stage, will xomni (only training vq and MLP, vit) and llava (only training MLP) have the same simple visual understanding capabilities?

**Comment by zhangxiaosong18** [comment]
In fact, we did not conduct evaluations at this stage 😂

**Comment by waltstephen** [comment]
Hahaha thank you, I can only wait until I train LLM (XD
